### PR TITLE
Faster tests

### PR DESCRIPTION
**Pull request by wmanley**
Speed up test runs by starting/stopping the service less.

All together these commits speed the tests up from ~270s to ~230s.  Still not that fast :(.

Here's a graph of the lifetimes of the test fixtures after these changes:

download

And before:

download 1

The graphs are not that useful in this instance but were helpful along the way.  See 550e51354eadbdc9e88fb712bb455bea89443c3a for the implementation.

Huh! and it turns out that github has support for rendering IPython Notebooks: https://github.com/wmanley/stb-tester-service/blob/550e51354eadbdc9e88fb712bb455bea89443c3a/profiling.ipynb

### CODE REVIEW

**Comment by drothlis** [comment]
> Huh! and it turns out that github has support for rendering IPython Notebooks

Nice.

File: http-service/service.py [hunk 1]
```diff
@@ -907,7 +907,8 @@ def press(self, key, remote_control=None, branch="master"):
                                   self.config['lirc'].get('driver'),
                                   self.config['lirc'].get('device'),
                                   self.config['lirc'].get(
-                                      'set_scheduling_priority', False)):
+                                      'set_scheduling_priority', False),
+                                  self.config['lirc'].get('binary')):
```
Comment (drothlis) [inline http-service/service.py]: Will this need a corresponding change to the stb-tester-one config?

File: tests/test_utils.py [hunk 2]
```diff
@@ -151,6 +162,10 @@ def get(self, endpoint, log_response=True, log_request=True,
 
     def post(self, endpoint, log_response=True, log_request=True,
              *args, **kwargs):
+        if self.readonly:
+            raise RuntimeError(
+                "Not allowed to mutate this TestRunner, please use a different "
+                "fixture in your test")
```
Comment (drothlis) [inline tests/test_utils.py]: Good idea.

**Comment by drothlis** [comment]
Finished reviewing. Looks good to me.

File: http-service/service.py (hunk 1 above)
Comment (wmanley) [inline http-service/service.py]: No, because it defaults to `"lircd"`

Comment (drothlis) [inline http-service/service.py]: But not when you explicitly pass in `None` as an argument, as you do here?

Comment (wmanley) [inline http-service/service.py]: Hmm, good point.  I just checked and it seems that we shouldn't be passing `None` in either.  We use `kwargs`.  See [`TestEnvironment`](https://github.com/wmanley/stb-tester-service/blob/6b8f6b5cae966252a9d64bd9630ccc5a31aa5c70/run-test-job.py#L341):

    dep = globals()[elemname](jobdir, **kwargs)

**Pull request merged**